dev-app/src/screens/HomeScreen.tsx

```diff
@@ -26,9 +28,48 @@ export default function HomeScreen() {
   const navigation = useNavigation();
   const { account } = useContext(AppContext);
   const [simulated, setSimulated] = useState<boolean>(true);
+  const [online, setOnline] = useState<boolean>(true);
   const [discoveryMethod, setDiscoveryMethod] =
     useState<Reader.DiscoveryMethod>('bluetoothScan');
-  const { disconnectReader, connectedReader } = useStripeTerminal();
+  const { disconnectReader, connectedReader } = useStripeTerminal({
+    onDidChangeOfflineStatus(status: OfflineStatus) {
+      console.log('offline status = ' + status.networkStatus);
+      setOnline(status.networkStatus == 'online' ? true : false);
+    },
+    onDidForwardingFailure(error) {
+      console.log('onDidForwardingFailure ' + error?.message);
+      let toast = Toast.show(error?.message ? error.message : 'unknown error', {
+        duration: Toast.durations.LONG,
+        position: Toast.positions.BOTTOM,
+        shadow: true,
+        animation: true,
+        hideOnPress: true,
+        delay: 0,
+      });
+
+      setTimeout(function () {
+        Toast.hide(toast);
+      }, 3000);
+    },
+    onDidForwardPaymentIntent(paymentIntent) {
```

Comment:
This callback also needs to accept an error. You can test this case by accepting an offline payment with a test amount that causes a decline:
https://stripe.com/docs/terminal/references/testing#physical-test-cards

The expectation is that `onDidForwardPaymentIntent` would be called back with an error indicating that the PI was declined.